File: src/test/java/edu/hw3/Task2Test.java
package edu.hw3;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import java.util.ArrayList;
import java.util.List;
import static org.assertj.core.api.Assertions.assertThat;

public class Task2Test {

    @Test
    @DisplayName("asda")
    void checl(){
        Task2 task = new Task2();

        List<String> out = new ArrayList<>(List.of("((()))", "(())", "()", "()", "(()())"));
        List<String> res = task.clusterize("((()))(())()()(()())");

        assertThat(res).isEqualTo(out);
    }
}

File: src/test/java/edu/hw3/Task2Test.kt
package edu.hw3

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class Task2Test {

    val task = Task2();

    @Test
    @DisplayName("Method should split round bracket expressions into clusters")
    fun check_clusterizing() {
        val originalExpression = "((()))(())()()(()())";


        val clusterizedExpression = task.clusterize(originalExpression);


        assertThat(clusterizedExpression).isEqualTo(listOf("((()))", "(())", "()", "()", "(()())"))
    }

}
